jobs/build/multi_promote/Jenkinsfile

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env groovy
+
+node {
+    checkout scm
+    def buildlib = load("pipeline-scripts/buildlib.groovy")
+    def commonlib = buildlib.commonlib
+    commonlib.describeJob("multi_promote", """
+        <h2>Kick off multiple promote jobs for selected nightlies</h2>
+    """)
+
+
+    def doozer_working = "${WORKSPACE}/doozer_working"
+
+    // Expose properties for a parameterized build
+    properties(
+        [
+            buildDiscarder(
+                logRotator(
+                    artifactDaysToKeepStr: '7',
+                    daysToKeepStr: '7'
+                )
+            ),
+            [
+                $class: 'ParametersDefinitionProperty',
+                parameterDefinitions: [
+                    choice(
+                        name: 'RELEASE_TYPE',
+                        description: 'Select [1. Standard Release] unless discussed with team lead',
+                        choices: [
+                                '1. Standard Release (Named, Signed, Previous, All Channels)',
+                                '2. Release Candidate (Named, Signed, Previous, Candidate Channel)',
+                                '3. Feature Candidate (Named, Signed - rpms may not be, Previous, Candidate Channel)',
+                                '4. Hotfix (No name, Signed, No Previous, All Channels)',
+                            ].join('\n'),
+                    ),
+                    string(
+                        name: "NIGHTLIES",
+                        description: "List of proposed nightlies for each arch, separated by comma",
+                        trim: true
+                    ),
+                    string(
+                        name: 'RELEASE_OFFSET',
+                        description: 'Integer. Do not specify for hotfix. If offset is X for 4.5 nightly => Release name is 4.5.X for standard, 4.5.0-rc.X for Release Candidate, 4.5.0-fc.X for Feature Candidate ',
+                        trim: true,
+                    ),
+                    commonlib.dryrunParam('Take no actions. Note: still notifies and runs signing job (which fails)'),
+                    commonlib.mockParam(),
+                ]
+            ],
+        ]
+    )
+
+    commonlib.checkMock()
+
+    // some basic validations
+    if (!params.NIGHTLIES) {
+        error("You must provide a list of proposed nightlies.")
+    }
+    
+    nightly_list = params.NIGHTLIES.split("[,\\s]+")
+    if (nightly_list.size() != 3) {
+        error("Something doesn't seem right. Job expects 3 nightlies of each arch")
+    }
```

Comment:
Thanks for pointing out, I wasn't aware of that. I think this is good for an initial draft, I'll write a separate story for this enhancement (also since I'm not entirely clear on mapping :))

Reply:
https://issues.redhat.com/browse/ART-2923